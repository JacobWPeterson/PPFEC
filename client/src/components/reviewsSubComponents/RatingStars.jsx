/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import styled from 'styled-components';

const StarsContainer = styled.div`
  unicode-bidi: bidi-override;
  color: #c5c5c5;
  font-size: 22px;
  height: 22px;
  line-height: 22px;
  position: relative;
  padding: 0;
  text-shadow: 0px 1px 0 #a2a2a2;

  &.top {
    width: ${({ rating }) => (rating / 5) * 100}%;
    color: #FDD700;
    padding: 0;
    position: absolute;
    z-index: 1;
    display: block;
    top: 0;
    left: 0;
    overflow: hidden;
  }

  &.bottom {
    padding: 0;
    display: block;
    z-index: 0;
  }

`;

const InputStarWrapper = styled.div`
  fontSize: 25px;
  color: #FDD700;
  cursor: pointer;
`;

class RatingStars extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedStars: [1, 2, 3, 4, 5],
      rating: this.props.rating || 0,
      hovered: 0,
      selectedIcon: '★',
      deselectedIcon: '☆',
    };
    this.setRating = this.setRating.bind(this);
    this.hoverRating = this.hoverRating.bind(this);
  }

  setRating(inputRating) {
    this.setState({
      rating: inputRating,
    });

    if (this.props.getRating) {
      this.props.getRating(this.state.rating);
    }
  }

  hoverRating(rating) {
    this.setState({
      hovered: rating,
    });
  }

  render() {
    let staticStars;
    let inputStars;
    if (this.props.rating) {
      staticStars = (
        <StarsContainer>
          <StarsContainer className="top" rating={this.props.rating}>
            <span>★</span>
            <span>★</span>
            <span>★</span>
            <span>★</span>
            <span>★</span>
          </StarsContainer>
          <StarsContainer className="bottom">
            <span>★</span>
            <span>★</span>
            <span>★</span>
            <span>★</span>
            <span>★</span>
          </StarsContainer>
        </StarsContainer>
      );
    } else {
      inputStars = (
        <InputStarWrapper>
          {this.state.selectedStars.map((star) => {
            const hoverStars = this.state.hovered < star
              ? this.state.deselectedIcon : this.state.selectedIcon;
            return (
              <span
                key={star}
                onClick={() => { this.setRating(star); }}
                onMouseEnter={() => { this.hoverRating(star); }}
                onMouseLeave={() => { this.hoverRating(0); }}
              >
                {this.state.rating < star ? hoverStars : this.state.selectedIcon}
              </span>
            );
          })}
        </InputStarWrapper>
      );
    }

    return (
      <div>
        {staticStars}
        {inputStars}
      </div>
    );
  }
}

export default RatingStars;
