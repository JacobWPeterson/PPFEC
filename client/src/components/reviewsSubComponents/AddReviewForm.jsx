import React from 'react';
import axios from 'axios';
import styled from 'styled-components';
import RatingStars from './RatingStars.jsx';

// product Charateristics
const productCharacteristics = {
  Size: ['A size too small', '½ a size too small', 'Perfect', '½ a size too big', 'A size too wide'],
  Width: ['Too narrow', 'Slightly narrow', 'Perfect', 'Slightly wide', 'Too wide'],
  Comfort: ['Uncomfortable', 'Slightly uncomfortable', 'Ok', 'Comfortable', 'Perfect'],
  Quality: ['Poor', 'Below average', 'What I expected', 'Pretty great', 'Perfect'],
  Length: ['Runs Short', 'Runs slightly short', 'Perfect', 'Runs slightly long', 'Runs long'],
  Fit: ['Runs tight', 'Runs slightly tight', 'Perfect', 'Runs slightly long', 'Runs long'],
};



const RatingWraper = styled.div`
  display: flex;
  align-items: baseline;
`;

const ValueButtonWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const CharateristicsSelectorWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: baseline;
`;

const StyledForm = styled.form`
  margin: 0 auto;
  padding: 20px;
  background-color: white;
  width: 800px;
  height: 100%;
`;

class AddReviewForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      productId: this.props.productId,
      rating: 0,
      recommend: true,
      summary: '',
      body: '',
      name: '',
      email: '',
    };
    this.getRating = this.getRating.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  handleSubmit(event) {
    console.log(this.state);
    axios.post();
    this.props.toggle();
    event.preventDefault();
  }

  getRating(inputRating) {
    this.setState({
      rating: inputRating,
    });
  }

  render() {
    const charateristicsKeys = Object.keys(this.props.characteristics);
    return (
      <StyledForm onSubmit={this.handleSubmit}>
        <h2>Write A Review</h2>
        <RatingWraper>
          <span>Overall Rating</span>
          <RatingStars getRating={this.getRating} />
        </RatingWraper>
        <div>
          <span>Do you recommend this product?</span>
          <input type="radio" name="recommend" value="true" required onChange={this.handleChange} />
          <label>Yes</label>
          <input type="radio" name="recommend" value="false" required onChange={this.handleChange} />
          <label>No</label>
        </div>
        <div>
          <span>Charateristics</span>
          {charateristicsKeys.map((key) => (
            <div key={key}>
              <div>{key}</div>
              <CharateristicsSelectorWrapper>
                {productCharacteristics[key].map((value, index) => (
                  <ValueButtonWrapper key={value}>
                    <span>{value}</span>
                    <input type="radio" name={this.props.characteristics[key].id} value={index + 1} required onChange={this.handleChange} />
                  </ValueButtonWrapper>
                ))}
              </CharateristicsSelectorWrapper>
            </div>
          ))}
        </div>
        <div>
          <label>Review Summary</label>
          <input type="text" id="reviewSummary" name="summary" cols="100" maxLength="60" placeholder="Example: Best purchase ever!" onChange={this.handleChange} />
        </div>
        <div>
          <label>Your Review</label>
          <div>
            <textarea id="reviewBody" name="body" cols="60" rows="10" maxLength="1000" placeholder="Why did you like the product or not?" required onChange={this.handleChange} />
          </div>
        </div>
        <div>
          <label>Upload the photos of your purchase</label>
          <div><input type="file" name="photos" value={this.state.photos} onChange={this.handleChange} /></div>
        </div>
        <div>
          <span>Your Nickname</span>
          <input type="text" id="reviewer_name" name="name" maxLength="60" placeholder="Example: jackson11!" required onChange={this.handleChange}/>
        </div>
        <div>
          <span>Your Email</span>
          <input type="email" id="email" name="email" maxLength="60" placeholder="Example: jackson11@mail.com" required onChange={this.handleChange}/>
        </div>
        <input type="submit" name="submit" />
        <button type="button" onClick={this.props.toggle}>Cancel</button>
      </StyledForm>

    );
  }
}

export default AddReviewForm;
