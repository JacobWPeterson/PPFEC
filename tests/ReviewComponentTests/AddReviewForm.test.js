/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import AddReviewForm from '../../client/src/components/reviewsSubComponents/AddReviewForm.jsx';
import RatingStars from '../../client/src/components/reviewsSubComponents/RatingStars.jsx';

describe('Add Review Form test', () => {
  const props = {
    summary: '',
    productId: 14950,
    characteristics: {
      Comfort: { id: 50085, value: '3.1891891891891892' },
      Fit: { id: 50083, value: '2.9189189189189189' },
      Length: { id: 50084, value: '2.5405405405405405' },
      Quality: { id: 50086, value: '3.1081081081081081' },
    },
  };

  const ratingStarsProps = {
    selectedStars: [1, 2, 3, 4, 5],
    rating: 4,
    hovered: 0,
    selectedIcon: '★',
    deselectedIcon: '☆',
  };
  let AddReviewFormComponent;
  let RatingStarsComponent;
  beforeEach(() => {
    AddReviewFormComponent = mount(<AddReviewForm {...props} debug />);
    RatingStarsComponent = mount(<RatingStars {...ratingStarsProps} debug />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correctly Add Review Form', () => {
    // mount render the whole dom with styled-component
    expect(AddReviewFormComponent).toMatchSnapshot();
  });

  it('The form should contain RatingStar component', () => {
    expect(AddReviewFormComponent.find(<RatingStars />)).toBeTruthy();
  });

  it('The characteristics rating section should has the same characteristics as the props', () => {
    expect(AddReviewFormComponent.find('Comfort')).toBeTruthy();
    expect(AddReviewFormComponent.find('Fit')).toBeTruthy();
    expect(AddReviewFormComponent.find('Length')).toBeTruthy();
    expect(AddReviewFormComponent.find('Quality')).toBeTruthy();
  });

  it('The form should capture summary input', () => {
    const summaryInput = AddReviewFormComponent.find('input').at(22);
    summaryInput.instance().value = 'summary testing';
    summaryInput.simulate('change');
    expect(AddReviewFormComponent.state('summary')).toBe('summary testing');
  });

  it('The form should have a textarea and it captures body input', () => {
    const bodyInput = AddReviewFormComponent.find('textarea').at(0);
    bodyInput.instance().value = 'body testing';
    bodyInput.simulate('change');
    expect(AddReviewFormComponent.state('body')).toBe('body testing');
  });

  it('The form should have a photo upload input', () => {
    const photoUpload = AddReviewFormComponent.find('input').at(23);
    const inputType = photoUpload.instance().type;
    expect(inputType).toBe('file');
  });

  it('The form should have a handle submit function', () => {
    const spyOnSubmit = jest.spyOn(AddReviewFormComponent.instance(), 'handleSubmit');
    const mockPreventDefault = jest.fn();
    const mockEvent = {
      preventDefault: mockPreventDefault,
    };
    AddReviewFormComponent.instance().handleSubmit(mockEvent);
    expect(spyOnSubmit).toBeCalled();
  });

  // it('Should have the correct rating when user click on the stars', () => {
  //   const spyOnRatingStarClick = jest.spyOn(RatingStarsComponent.instance(), 'setRating');
  //   const mockPreventDefault = jest.fn();

  //   expect(AddReviewFormComponent.state('rating')).toBe(ratingStarsProps.rating);
  // });
});
