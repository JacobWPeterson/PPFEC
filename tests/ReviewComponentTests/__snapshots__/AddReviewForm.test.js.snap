// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Review Form test Should render correctly Add Review Form 1`] = `
<AddReviewForm
  characteristics={
    Object {
      "Comfort": Object {
        "id": 50085,
        "value": "3.1891891891891892",
      },
      "Fit": Object {
        "id": 50083,
        "value": "2.9189189189189189",
      },
      "Length": Object {
        "id": 50084,
        "value": "2.5405405405405405",
      },
      "Quality": Object {
        "id": 50086,
        "value": "3.1081081081081081",
      },
    }
  }
  debug={true}
  productId={14950}
  summary=""
>
  <AddReviewForm__StyledForm
    onSubmit={[Function]}
  >
    <form
      className="AddReviewForm__StyledForm-kljszx-6 ifPRRz"
      onSubmit={[Function]}
    >
      <h2>
        Write A Review
      </h2>
      <AddReviewForm__VerticalWrapper>
        <div
          className="AddReviewForm__VerticalWrapper-kljszx-0 NSeLo"
        >
          <AddReviewForm__StyledLabel>
            <span
              className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
            >
              Overall Rating
            </span>
          </AddReviewForm__StyledLabel>
          <RatingStars
            getRating={[Function]}
          >
            <div>
              <RatingStars__InputStarWrapper>
                <div
                  className="RatingStars__InputStarWrapper-sc-1djn8fm-1 kKUgkY"
                >
                  <span
                    key="1"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    ☆
                  </span>
                  <span
                    key="2"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    ☆
                  </span>
                  <span
                    key="3"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    ☆
                  </span>
                  <span
                    key="4"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    ☆
                  </span>
                  <span
                    key="5"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    ☆
                  </span>
                </div>
              </RatingStars__InputStarWrapper>
            </div>
          </RatingStars>
        </div>
      </AddReviewForm__VerticalWrapper>
      <AddReviewForm__VerticalWrapper>
        <div
          className="AddReviewForm__VerticalWrapper-kljszx-0 NSeLo"
        >
          <AddReviewForm__StyledLabel>
            <span
              className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
            >
              Do you recommend this product?
            </span>
          </AddReviewForm__StyledLabel>
          <div>
            <input
              name="recommend"
              onChange={[Function]}
              required={true}
              type="radio"
              value="true"
            />
            <span>
              Yes
            </span>
            <input
              name="recommend"
              onChange={[Function]}
              required={true}
              type="radio"
              value="false"
            />
            <span>
              No
            </span>
          </div>
        </div>
      </AddReviewForm__VerticalWrapper>
      <div>
        <AddReviewForm__StyledLabel>
          <span
            className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
          >
            Charateristics
          </span>
        </AddReviewForm__StyledLabel>
        <ul
          style={
            Object {
              "margin": "0 auto",
              "padding": "10px",
            }
          }
        >
          <li
            key="Comfort"
          >
            <AddReviewForm__StyledLabel>
              <span
                className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
              >
                Comfort
              </span>
            </AddReviewForm__StyledLabel>
            <AddReviewForm__CharateristicsSelectorWrapper>
              <div
                className="AddReviewForm__CharateristicsSelectorWrapper-kljszx-5 kwtDeo"
              >
                <AddReviewForm__ValueButtonWrapper
                  key="Uncomfortable"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Uncomfortable
                    </span>
                    <input
                      name={50085}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={1}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Slightly uncomfortable"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Slightly uncomfortable
                    </span>
                    <input
                      name={50085}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={2}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="OK"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      OK
                    </span>
                    <input
                      name={50085}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={3}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Comfortable"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Comfortable
                    </span>
                    <input
                      name={50085}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={4}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Perfect"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Perfect
                    </span>
                    <input
                      name={50085}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={5}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
              </div>
            </AddReviewForm__CharateristicsSelectorWrapper>
          </li>
          <li
            key="Fit"
          >
            <AddReviewForm__StyledLabel>
              <span
                className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
              >
                Fit
              </span>
            </AddReviewForm__StyledLabel>
            <AddReviewForm__CharateristicsSelectorWrapper>
              <div
                className="AddReviewForm__CharateristicsSelectorWrapper-kljszx-5 kwtDeo"
              >
                <AddReviewForm__ValueButtonWrapper
                  key="Runs tight"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs tight
                    </span>
                    <input
                      name={50083}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={1}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Runs slightly tight"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs slightly tight
                    </span>
                    <input
                      name={50083}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={2}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Perfect"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Perfect
                    </span>
                    <input
                      name={50083}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={3}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Runs slightly long"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs slightly long
                    </span>
                    <input
                      name={50083}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={4}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Runs long"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs long
                    </span>
                    <input
                      name={50083}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={5}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
              </div>
            </AddReviewForm__CharateristicsSelectorWrapper>
          </li>
          <li
            key="Length"
          >
            <AddReviewForm__StyledLabel>
              <span
                className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
              >
                Length
              </span>
            </AddReviewForm__StyledLabel>
            <AddReviewForm__CharateristicsSelectorWrapper>
              <div
                className="AddReviewForm__CharateristicsSelectorWrapper-kljszx-5 kwtDeo"
              >
                <AddReviewForm__ValueButtonWrapper
                  key="Runs Short"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs Short
                    </span>
                    <input
                      name={50084}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={1}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Runs slightly short"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs slightly short
                    </span>
                    <input
                      name={50084}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={2}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Perfect"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Perfect
                    </span>
                    <input
                      name={50084}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={3}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Runs slightly long"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs slightly long
                    </span>
                    <input
                      name={50084}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={4}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Runs long"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Runs long
                    </span>
                    <input
                      name={50084}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={5}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
              </div>
            </AddReviewForm__CharateristicsSelectorWrapper>
          </li>
          <li
            key="Quality"
          >
            <AddReviewForm__StyledLabel>
              <span
                className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
              >
                Quality
              </span>
            </AddReviewForm__StyledLabel>
            <AddReviewForm__CharateristicsSelectorWrapper>
              <div
                className="AddReviewForm__CharateristicsSelectorWrapper-kljszx-5 kwtDeo"
              >
                <AddReviewForm__ValueButtonWrapper
                  key="Poor"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Poor
                    </span>
                    <input
                      name={50086}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={1}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Below average"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Below average
                    </span>
                    <input
                      name={50086}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={2}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="What I expected"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      What I expected
                    </span>
                    <input
                      name={50086}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={3}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Pretty great"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Pretty great
                    </span>
                    <input
                      name={50086}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={4}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
                <AddReviewForm__ValueButtonWrapper
                  key="Perfect"
                >
                  <div
                    className="AddReviewForm__ValueButtonWrapper-kljszx-3 camWqr"
                  >
                    <span>
                      Perfect
                    </span>
                    <input
                      name={50086}
                      onChange={[Function]}
                      required={true}
                      type="radio"
                      value={5}
                    />
                  </div>
                </AddReviewForm__ValueButtonWrapper>
              </div>
            </AddReviewForm__CharateristicsSelectorWrapper>
          </li>
        </ul>
      </div>
      <AddReviewForm__VerticalWrapper>
        <div
          className="AddReviewForm__VerticalWrapper-kljszx-0 NSeLo"
        >
          <AddReviewForm__StyledLabel>
            <span
              className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
            >
              Review Summary
            </span>
          </AddReviewForm__StyledLabel>
          <AddReviewForm__StyledInput
            cols="100"
            id="reviewSummary"
            maxLength="60"
            name="summary"
            onChange={[Function]}
            placeholder="Example: Best purchase ever!"
            type="text"
            value=""
          >
            <input
              className="AddReviewForm__StyledInput-kljszx-10 folYNU"
              cols="100"
              id="reviewSummary"
              maxLength="60"
              name="summary"
              onChange={[Function]}
              placeholder="Example: Best purchase ever!"
              type="text"
              value=""
            />
          </AddReviewForm__StyledInput>
        </div>
      </AddReviewForm__VerticalWrapper>
      <AddReviewForm__VerticalWrapper>
        <div
          className="AddReviewForm__VerticalWrapper-kljszx-0 NSeLo"
        >
          <AddReviewForm__StyledLabel>
            <span
              className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
            >
              Your Review
            </span>
          </AddReviewForm__StyledLabel>
          <AddReviewForm__StyledTextarea
            cols="60"
            id="reviewBody"
            maxLength="1000"
            name="body"
            onChange={[Function]}
            placeholder="Why did you like the product or not?"
            required={true}
            rows="10"
            value=""
          >
            <textarea
              className="AddReviewForm__StyledTextarea-kljszx-11 KyGPD"
              cols="60"
              id="reviewBody"
              maxLength="1000"
              name="body"
              onChange={[Function]}
              placeholder="Why did you like the product or not?"
              required={true}
              rows="10"
              value=""
            />
          </AddReviewForm__StyledTextarea>
        </div>
      </AddReviewForm__VerticalWrapper>
      <AddReviewForm__HorizontalWrapper>
        <div
          className="AddReviewForm__HorizontalWrapper-kljszx-1 kHOVap"
        >
          <AddReviewForm__StyledLabel>
            <span
              className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
            >
              Upload the photos of your purchase
            </span>
          </AddReviewForm__StyledLabel>
          <input
            multiple={true}
            name="photos"
            onChange={[Function]}
            type="file"
          />
        </div>
      </AddReviewForm__HorizontalWrapper>
      <AddReviewForm__NameEmailWraper>
        <div
          className="AddReviewForm__NameEmailWraper-kljszx-2 bQdAvP"
        >
          <AddReviewForm__VerticalWrapper>
            <div
              className="AddReviewForm__VerticalWrapper-kljszx-0 NSeLo"
            >
              <AddReviewForm__StyledLabel>
                <span
                  className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
                >
                  Your Nickname
                </span>
              </AddReviewForm__StyledLabel>
              <AddReviewForm__NameEmailInput
                id="reviewer_name"
                maxLength="60"
                name="name"
                onChange={[Function]}
                placeholder="Example: jackson11!"
                required={true}
                type="text"
                value=""
              >
                <input
                  className="AddReviewForm__NameEmailInput-kljszx-7 iLzyob"
                  id="reviewer_name"
                  maxLength="60"
                  name="name"
                  onChange={[Function]}
                  placeholder="Example: jackson11!"
                  required={true}
                  type="text"
                  value=""
                />
              </AddReviewForm__NameEmailInput>
            </div>
          </AddReviewForm__VerticalWrapper>
          <AddReviewForm__VerticalWrapper>
            <div
              className="AddReviewForm__VerticalWrapper-kljszx-0 NSeLo"
            >
              <AddReviewForm__StyledLabel>
                <span
                  className="AddReviewForm__StyledLabel-kljszx-4 bIiwfI"
                >
                  Your Email
                </span>
              </AddReviewForm__StyledLabel>
              <AddReviewForm__NameEmailInput
                id="email"
                maxLength="60"
                name="email"
                onChange={[Function]}
                placeholder="Example: jackson11@mail.com"
                required={true}
                type="email"
                value=""
              >
                <input
                  className="AddReviewForm__NameEmailInput-kljszx-7 iLzyob"
                  id="email"
                  maxLength="60"
                  name="email"
                  onChange={[Function]}
                  placeholder="Example: jackson11@mail.com"
                  required={true}
                  type="email"
                  value=""
                />
              </AddReviewForm__NameEmailInput>
            </div>
          </AddReviewForm__VerticalWrapper>
        </div>
      </AddReviewForm__NameEmailWraper>
      <AddReviewForm__SytledSubmitButton
        name="submit"
        type="submit"
      >
        <input
          className="AddReviewForm__SytledSubmitButton-kljszx-8 cxbQoL"
          name="submit"
          type="submit"
        />
      </AddReviewForm__SytledSubmitButton>
      <AddReviewForm__SytledButton
        type="button"
      >
        <button
          className="AddReviewForm__SytledButton-kljszx-9 bXkUT"
          type="button"
        >
          Cancel
        </button>
      </AddReviewForm__SytledButton>
    </form>
  </AddReviewForm__StyledForm>
</AddReviewForm>
`;
